src/vsr/replica.zig

```diff
@@ -7840,6 +7824,15 @@ pub fn ReplicaType(
             assert(self.log_view_durable() <= self.view_durable());
             assert(self.log_view_durable() <= self.log_view);
 
+            switch (self.syncing) {
+                .updating_superblock => |stage| {
+                    if (stage.checkpoint_state.header.op == self.op_checkpoint()) {
```

Comment:
There might be several updates queued, so we check that the correct one finished